src/App.jsx

```diff
@@ -62,57 +230,57 @@ const RPGInterface = () => {
         </div>
 
         {/* Health Bars */}
-        {!menu && (
+        {!(gameState.gameScene === GameScene.MENU_SCENE) && (
           <div className="grid grid-cols-1 md:grid-cols-2 gap-4 mb-8 space-x">
-            <HealthBar 
-              current={playerHealth} 
-              max={playerMaxHealth} 
-              label="Player Health" 
+            <HealthBar
+              current={gameState.player.shield}
+              max={playerShieldMax}
+              label="Player Health"
               color="bg-green-500"
             />
-            <HealthBar 
-              current={enemyHealth} 
-              max={enemyMaxHealth} 
-              label="Enemy Health" 
+            <HealthBar
+              current={gameState.enemy === null ? 0 : gameState.enemy.shield}
+              max={enemyShieldMax}
+              label="Enemy Health"
               color="bg-red-800"
             />
           </div>
         )}
 
-          {/* Command Buttons */}
-          <div className="space-y-4">
-            <div className="bg-gray-800 rounded-lg border border-gray-700 p-4">
-              <h3 className="text-sm font-medium text-gray-300 mb-4">{!menu ? 'Combat Actions' : 'Game Options'}</h3>
-              <div className="sm:grid sm:grid-cols-2 sm:gap-4 ">
-              {!menu ? (
+        {/* Command Buttons */}
+        <div className="space-y-4">
+          <div className="bg-gray-800 rounded-lg border border-gray-700 p-4">
+            <h3 className="text-sm font-medium text-gray-300 mb-4">{!(gameState.gameScene === GameScene.MENU_SCENE) ? 'Combat Actions' : 'Game Options'}</h3>
+            <div className="sm:grid sm:grid-cols-2 sm:gap-4 ">
+              {!(gameState.gameScene === GameScene.MENU_SCENE) ? (
                 <>
-                  <button className="block w-full sm:w-auto mb-4 sm:mb-0 bg-red-800 hover:bg-red-900 text-white font-medium py-2 px-4 rounded-lg transition-colors duration-200">
+                  <button className="block w-full sm:w-auto mb-4 sm:mb-0 bg-red-800 hover:bg-red-900 text-white font-medium py-2 px-4 rounded-lg transition-colors duration-200" onClick={() => { setGameState(prevState => (handleAction(prevState, BattleSceneAction.ATTACK))); }}>
```

Comment:
```suggestion
                  <button className="block w-full sm:w-auto mb-4 sm:mb-0 bg-red-800 hover:bg-red-900 text-white font-medium py-2 px-4 rounded-lg transition-colors duration-200" onClick={() => handleAction(BattleSceneAction.ATTACK)}>
```
See my refactoring of the `handleAction` above.